tests/unit/amo/components/TestAddon.js

```diff
@@ -748,6 +787,284 @@ describe('Addon', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeAddon,
+          type: ADDON_TYPE_OPENSEARCH,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More add-ons by Krupa');
+    });
+
+    it('displays a plural form when extension has multiple authors', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeAddon,
+          authors: Array(2).fill(fakeAddon.authors[0]),
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More extensions by these developers');
+    });
+
+    it('displays a plural form when theme has multiple authors', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeTheme,
+          authors: Array(2).fill(fakeTheme.authors[0]),
+        },
+        addonsByAuthors: [fakeTheme],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More themes by these artists');
+    });
+
+    it('displays a plural form when language pack has multiple authors', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeAddon,
+          authors: Array(2).fill(fakeAddon.authors[0]),
+          type: ADDON_TYPE_LANG,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More language packs by these translators');
+    });
+
+    it('displays a plural form when dictionary has multiple authors', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeAddon,
+          authors: Array(2).fill(fakeAddon.authors[0]),
+          type: ADDON_TYPE_DICT,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More dictionaries by these translators');
+    });
+
+    it('displays a plural form when add-on has multiple authors', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeAddon,
+          authors: Array(2).fill(fakeAddon.authors[0]),
+          type: ADDON_TYPE_OPENSEARCH,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More add-ons by these developers');
+    });
+
+    it('displays more add-ons by authors', () => {
+      const { store } = dispatchClientMetadata();
+
+      const addon = fakeAddon;
+      const addonsByAuthors = [
+        { ...fakeAddon, slug: 'addon-1' },
+        { ...fakeAddon, slug: 'addon-2' },
+        { ...fakeAddon, slug: 'addon-3' },
+      ];
+
+      store.dispatch(_loadAddon({ addon }));
+      store.dispatch(_loadOtherAddonsByAuthors({ addon, addonsByAuthors }));
+
+      const mappedProps = mapStateToProps(
```

Comment:
This approach looks good but I think you should take this opportunity to render it properly with `shallowUntilTarget`:

```diff
diff --git a/tests/unit/amo/components/TestAddon.js b/tests/unit/amo/components/TestAddon.js
index 3c04fe9c..bcc2ad55 100644
--- a/tests/unit/amo/components/TestAddon.js
+++ b/tests/unit/amo/components/TestAddon.js
@@ -10,7 +10,7 @@ import { Provider } from 'react-redux';
 import { match } from 'react-router';
 
 import { setViewContext } from 'amo/actions/viewContext';
-import {
+import Addon, {
   AddonBase,
   mapStateToProps,
 } from 'amo/components/Addon';
@@ -58,6 +58,7 @@ import {
   createStubErrorHandler,
   getFakeI18nInst,
   sampleUserAgentParsed,
+  shallowUntilTarget,
 } from 'tests/unit/helpers';
 import ErrorList from 'ui/components/ErrorList';
 import LoadingText from 'ui/components/LoadingText';
@@ -107,6 +108,11 @@ function renderAsDOMNode(...args) {
   return findDOMNode(root);
 }
 
+function renderComponent(...args) {
+  const props = renderProps(...args);
+  return shallowUntilTarget(<Addon {...props} />, AddonBase);
+}
+
 function shallowRender(...args) {
   const props = renderProps(...args);
   return shallow(<AddonBase {...props} />, { context: { i18n: props.i18n } });
@@ -976,11 +982,9 @@ describe(__filename, () => {
       store.dispatch(_loadAddon({ addon }));
       store.dispatch(_loadOtherAddonsByAuthors({ addon, addonsByAuthors }));
 
-      const mappedProps = mapStateToProps(
-        store.getState(), { params: { slug: addon.slug } }
-      );
-
-      const root = shallowRender({ ...mappedProps });
+      const root = renderComponent({
+        params: { slug: addon.slug }, store,
+      });
 
       expect(root.find('.AddonDescription-more-addons--theme'))
         .toHaveLength(0);
@@ -1001,11 +1005,9 @@ describe(__filename, () => {
       store.dispatch(_loadAddon({ addon }));
       store.dispatch(_loadOtherAddonsByAuthors({ addon, addonsByAuthors }));
 
-      const mappedProps = mapStateToProps(
-        store.getState(), { params: { slug: addon.slug } }
-      );
-
-      const root = shallowRender({ ...mappedProps });
+      const root = renderComponent({
+        params: { slug: addon.slug }, store,
+      });
 
       expect(root.find('.AddonDescription-more-addons--theme'))
         .toHaveLength(1);
```

Reply:
thanks for the patch!